src/ui_ng/lib/src/datetime-picker/datetime-picker.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, Input, Output, EventEmitter, ViewChild } from '@angular/core';
+import { NgModel } from '@angular/forms';
+
+import { DATETIME_PICKER_TEMPLATE } from './datetime-picker.component.html';
+
+@Component({
+  selector: 'hbr-datetime',
+  template: DATETIME_PICKER_TEMPLATE
+})
+export class DatePickerComponent {
+  
+  @Input() dateInput: string;
+  @Input() oneDayOffset: boolean;
+
+  @ViewChild('searchTime') 
+  searchTime: NgModel;
+
+  @Output() search = new EventEmitter<string>();
+
+  get dateInvalid(): boolean {
+    return (this.searchTime.errors && this.searchTime.errors.dateValidator && (this.searchTime.dirty || this.searchTime.touched)) || false;
+  }
+
+  convertDate(strDate: string): string {
```

Comment:
Is this function safe to cover all the situations? Why don't you create a Date object and use format to do the conversation ?